collie/model/hybrid_matrix_factorization.py

```diff
@@ -412,6 +405,43 @@ def _setup_model(self, **kwargs) -> None:
             nn.init.xavier_normal_(self.combined_layers[i].weight)
             self.add_module('combined_layer_{}'.format(i), layer)
 
+    def _calculate_metadata_output(
+        self,
+        metadata_type: str,
+        users: torch.tensor,
+        items: torch.tensor,
+    ) -> None:
+        """
+        Calculate metadata output for either item or user data.
+
+        Parameters
+        ----------
+        metadata_type: str
+            Metadata type, one of ``user`` or ``item``
+        users: tensor, 1-d
+            Array of user indices
+        items: tensor, 1-d
+            Array of item indices
+
+        """
+        # TODO: remove self.device and let lightning do it
+
+        metadata = getattr(self, f'{metadata_type}_metadata')
+        metadata_layers = getattr(self, f'{metadata_type}_metadata_layers')
+        if metadata_type == 'item':
+            metadata_output = metadata[items, :].to(self.device)
+        else:
+            metadata_output = metadata[users, :].to(self.device)
+        if metadata_layers is not None:
```

Comment:
[Optional] Can we add a space here to make it clear this `if` block is separate from the one above?

Reply:
👍